Propogate batch id

bio-boris opened the pull request:
Fix missing batch id and update test
Update htcondor per broken build https://github.com/kbase/execution_engine2/pull/410/checks?check_run_id=3136248755

MrCreosote reviewed (approved):
LGTM other than the tests aren't passing

bio-boris commented:
@MrCreosote I went and deleted all of the version numbers from requirements.txt
 Then I ran a new build and a new pip freeze. I updated the file here https://github.com/kbase/execution_engine2/pull/410/files
Now the tests are passing
